drivers/metadata/metadata.go

```diff
@@ -172,25 +172,47 @@ func NewCatalogSet(v []Catalog) *CatalogSet {
 		r[i] = &v[i]
 	}
 	return &CatalogSet{
-		resultSet: resultSet{
+		resultSet: &resultSet{
 			results: r,
 			columns: []string{"Catalog"},
 		},
 	}
 }
 
+func NewCatalogSetWithColumns(v []Result, cols []string) *CatalogSet {
+	return &CatalogSet{
+		resultSet: &resultSet{
+			results: v,
+			columns: cols,
+		},
+	}
+}
+
+type CatalogInterface interface {
+	GetCatalog() *Catalog
+}
+
 func (s CatalogSet) Get() *Catalog {
-	return s.results[s.current-1].(*Catalog)
+	r := s.results[s.current-1]
+	catalog, ok := r.(CatalogInterface)
+	if ok {
+		return catalog.GetCatalog()
+	}
+	return nil
```

Comment:
It's wrong to return a nil here and I think it's ok to require all custom results to implement this interface.
```suggestion
	return r.(CatalogProvider).GetCatalog()
```